common/djangoapps/entitlements/api/v1/views.py

```diff
@@ -0,0 +1,53 @@
+import logging
+
+from django.utils import timezone
+from django_filters.rest_framework import DjangoFilterBackend
+from edx_rest_framework_extensions.authentication import JwtAuthentication
+from rest_framework import permissions, viewsets
+from rest_framework.authentication import SessionAuthentication
+
+from entitlements.api.v1.filters import CourseEntitlementFilter
+from entitlements.models import CourseEntitlement
+from entitlements.serializers import CourseEntitlementSerializer
+from student.models import CourseEnrollment
+
+log = logging.getLogger(__name__)
+
+
+class EntitlementViewSet(viewsets.ModelViewSet):
+    authentication_classes = (JwtAuthentication, SessionAuthentication,)
+    permission_classes = (permissions.IsAuthenticated, permissions.IsAdminUser,)
+    queryset = CourseEntitlement.objects.all().select_related('user')
```

Comment:
Do you also want to select the enrollment?

Reply:
I do not think it is necessary right now as I do not return anything about the Course Run.